src/pypromice/L0toL1.py

```diff
@@ -121,62 +124,322 @@ def toL1(L0, vars_df, flag_file=None, T_0=273.15, tilt_threshold=-100):
             ds['wspd_x_i'], ds['wspd_y_i'] = _calcWindDir(ds['wspd_i'], ds['wdir_i'])   
     return ds
 
-def _flagNAN(ds, flag_file=None):
+def _getDF(flag_url, flag_file, download=True):
+    '''Get dataframe from flag or adjust file. First attempt to retrieve from 
+    URL. If this fails then attempt to retrieve from local file 
+    
+    Parameters
+    ----------
+    flag_url : str
+        URL address to file
+    flag_file : str
+        Local path to file
+    
+    Returns
+    -------
+    df : pd.DataFrame
+        Flag or adjustment dataframe
+    '''
+    # Attempt to retrieve URL csv as DataFrame
+    try:
+        df = pd.read_csv(
+                        flag_url, 
+                        comment="#", 
+                        skipinitialspace=True,
+                        ).dropna(how='all', axis='rows')
+        print(f'{flag_url} retrieved')
+        
+        if download==True and os.path.isdir(os.path.dirname(flag_file))==True:
+            df.to_csv(flag_file)
+            print(f'File saved to {flag_file}')
+    
+    # Attempt to retrieve local csv as DataFrame
+    except HTTPError:
```

Comment:
If I turn off my internet connectivity entirely, and try to run:
```
In [9]: foo = pd.read_csv('https://raw.githubusercontent.com/GEUS-Glaciology-and-Climate/PROMICE-AWS-data-issues/master/flags/NSE.csv')
```
I get `URLError`. So we need to `except` and `import` that as well.

`except (HTTPError, URLError):`

Reply:
Ah gotcha.

Reply:
Added now.